extensions/applicationinsights-analytics-js/Tests/Unit/src/AnalyticsPlugin.tests.ts

```diff
@@ -1063,324 +1063,329 @@ export class AnalyticsPluginTests extends AITestClass {
-                };
-                this.sandbox.spy()
-                this.trackSpy = this.sandbox.spy(sender, "_onSuccess");
+                    this.throwInternalSpy = this.sandbox.spy(appInsights.core.logger, "throwInternal");
+                    sender._sender = (payload:any[], isAsync:boolean) => {
+                        sender._onSuccess(payload, payload.length);
+                    };
+                    this.sandbox.spy()
+                    this.trackSpy = this.sandbox.spy(sender, "_onSuccess");
 
-                this.exceptionHelper.capture(appInsights);
+                    this.exceptionHelper.capture(appInsights);
 
-                this.causeException(() => {
-                    this.exceptionHelper.throwRuntimeException(AITestClass.orgSetTimeout);
-                });
+                    this.causeException(() => {
+                        this.exceptionHelper.throwRuntimeException(AITestClass.orgSetTimeout);
+                    });
 
-                Assert.ok(!this.trackSpy.calledOnce, "track not called yet");
-                Assert.ok(!this.throwInternalSpy.called, "No internal errors");
-            }].concat(this.waitForException(1))
-            .concat(() => {
-                let isLocal = window.location.protocol === "file:";
-                let exp = this.trackSpy.args[0];
-                const payloadStr: string[] = this.getPayloadMessages(this.trackSpy);
-            
-                if (payloadStr.length > 0) {
-                    const payload = JSON.parse(payloadStr[0]);
-                    const data = payload.data;
-                    Assert.ok(data, "Has Data");
-                    if (data) {
-                        Assert.ok(data.baseData, "Has BaseData");
-                        let baseData = data.baseData;
-                        if (baseData) {
-                            const ex = baseData.exceptions[0];
-                            if (isLocal) {
-                                Assert.ok(ex.message.indexOf("Script error:") !== -1, "Make sure the error message is present [" + ex.message + "]");
-                                Assert.equal("String", ex.typeName, "Got the correct typename [" + ex.typeName + "]");
-                            } else {
-                                Assert.ok(ex.message.indexOf("ug is not a function") !== -1, "Make sure the error message is present [" + ex.message + "]");
-                                Assert.equal("TypeError", ex.typeName, "Got the correct typename [" + ex.typeName + "]");
-                                Assert.ok(baseData.properties["columnNumber"], "has column number");
-                                Assert.ok(baseData.properties["lineNumber"], "has Line number");
+                    Assert.ok(!this.trackSpy.calledOnce, "track not called yet");
+                    Assert.ok(!this.throwInternalSpy.called, "No internal errors");
+                })
+                .concat(this.waitForException(1))
```

Comment:
The waitForException is returning an array of steps, that will need to be changed to return a promise

Reply:
If the previous function is no longer used then remove it